pkg/api/mlflow/service/metric/service.go

```diff
@@ -12,24 +12,37 @@ import (
 
 // Service provides service layer to work with `metric` business logic.
 type Service struct {
+	runRepository    repositories.RunRepositoryProvider
 	metricRepository repositories.MetricRepositoryProvider
 }
 
 // NewService creates new Service instance.
-func NewService(metricRepository repositories.MetricRepositoryProvider) *Service {
+func NewService(
+	runRepository repositories.RunRepositoryProvider,
+	metricRepository repositories.MetricRepositoryProvider,
+) *Service {
 	return &Service{
+		runRepository:    runRepository,
 		metricRepository: metricRepository,
 	}
 }
 
 func (s Service) GetMetricHistory(
-	ctx context.Context, req *request.GetMetricHistoryRequest,
+	ctx context.Context, namespace *models.Namespace, req *request.GetMetricHistoryRequest,
```

Comment:
I think the argument types feel a little mismatched -- one is a `model`, one a `request`. On the one hand, it seems like namespace could just be an attribute on the GetMetricHistoryRequest, but I also don't quite think the controller-level request should be used in the service (similarly, the controller-level api errors feel like the wrong level).